libs/grok/src/test/java/org/elasticsearch/grok/PatternBankTests.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.grok;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class PatternBankTests extends ESTestCase {
+
+    public void testInternalBankIsUnmodifiableAndACopy() {
+        Map<String, String> bank = new HashMap<>();
+        bank.put("ONE", "1");
+        var patternBank = new PatternBank(bank);
+        assertNotSame(patternBank.bank(), bank);
+        assertEquals(patternBank.bank(), bank);
```

Comment:
Can we do a quick expectThrows here to make sure we can't set a value in the bank?

Reply:
added thanks!